internal/bgp/bgpfrr/config.go

```diff
@@ -0,0 +1,127 @@
+package bgpfrr
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"text/template"
+)
+
+// TODO configFileName will need to be set to the filename of the FRR configuration
+// file in the container.
+var configFileName = "frr.conf"
+
+const configTemplate = `
+log stdout {{.Loglevel}}
+hostname {{.Hostname}}
+
+{{range .Routers -}}
+router bgp {{.Asn}}
+  no bgp default ipv4-unicast
+  bgp router-id {{.RouterId}}
+{{range .Neighbors }}
+  neighbor {{.Addr}} remote-as {{.Asn}}
+{{- end }}
+{{range $n := .Neighbors -}}
+{{range .Advertisements }}
+  address-family {{.Version}} unicast
+    neighbor {{$n.Addr}} activate
+    network {{.Prefix}}
+  exit-address-family
+{{- end}}
+{{end}}
+{{end}}
+`
```

Comment:
this good start but I would to see further research on how best to represent BGP configs in FRR terms, the other day I was checking there peer groups , route-map, bgp list and the list just goes on

Reply:
Thanks for the review @msherif1234. Yes, I assume there is a more optimal way to do this translation but I am not an expert on FRR so I don't know what it is. Even if this is non-optimal, it should server as a starting point and as we integrate into MetalLB and develop tests, we should be able to improve it.

I did consider netsted-templates while I was developing this but didn't think they were warranted at this stage. However, I will have another look and see if they will simplify things for us now.

Reply:
it will be great if we have main template and that can invoke per AF template and the AF template invoke peer template what do you think ?

Reply:
I started to look at this. There are a number of ways we could arrange the templates. However, at this stage, I don't think we know what the final configuration file should look like. Therefore, I suggest that we wait until we have merged this and integrated with FRR through a CI and we can refactor after this. I will add a note in the code outlining this suggestion. How does that sound?